ec/src/hashing/curve_maps/wb/mod.rs

```diff
@@ -32,18 +32,48 @@ pub struct IsogenyMap<
     DOMAIN: SWCurveConfig,
     CODOMAIN: SWCurveConfig<BaseField = BaseField<DOMAIN>>,
 > {
-    pub x_map_numerator: &'a [BaseField<DOMAIN>],
+    pub x_map_numerator: &'a [BaseField<CODOMAIN>],
     pub x_map_denominator: &'a [BaseField<CODOMAIN>],
 
-    pub y_map_numerator: &'a [BaseField<DOMAIN>],
+    pub y_map_numerator: &'a [BaseField<CODOMAIN>],
     pub y_map_denominator: &'a [BaseField<CODOMAIN>],
+    _phantom_domain: PhantomData<DOMAIN>,
 }
 
+// Pratyush suggestion
+// impl</* bounds */> IsogenyMap<'a, DOMAIN, CODOMAIN> {
+// 	const fn new(
+// 	    x_numerator: &'a [BaseField<CODOMAIN>],
+// 	    x_denominator: &'a [BaseField<CODOMAIN>],
+// 	    ...
+// 	) → Self {
+// 	    Self {
+// 	        x_numerator,
+// 	        x_denominator,
+// 	        ...
+// 	    }
+// 	}
+// }
 impl<'a, DOMAIN, CODOMAIN> IsogenyMap<'a, DOMAIN, CODOMAIN>
 where
     DOMAIN: SWCurveConfig,
     CODOMAIN: SWCurveConfig<BaseField = BaseField<DOMAIN>>,
 {
+    pub const fn new(
+        x_map_numerator: &'a [BaseField<CODOMAIN>],
+        x_map_denominator: &'a [BaseField<CODOMAIN>],
+        y_map_numerator: &'a [BaseField<CODOMAIN>],
+        y_map_denominator: &'a [BaseField<CODOMAIN>],
+    ) -> Self {
+        Self {
+            x_map_numerator,
+            x_map_denominator,
+            y_map_numerator,
+            y_map_denominator,
+            _phantom_domain: PhantomData::<DOMAIN>,
+        }
+    }
```

Comment:
We can also limit this to `'static` lifetimes. That might help with the "temporary dropped while borrowed" issue.

Reply:
do you mean to hard code static life time. I think I did that and the error still persists.

Reply:
I tried it too, couldn't find a way around it...